c_glib/test/gandiva/test-function-registry.rb

```diff
@@ -19,6 +19,7 @@ class TestGandivaFunctionRegistry < Test::Unit::TestCase
   include Helper::DataType
 
   def setup
+    omit("Skip Gandiva FunctionRegistry tests because the underlying API is changed")
```

Comment:
> So we can ignore c_glib related test failures for now

@kou I skip the unit tests in `test-function-registry.rb` and `test-native-function.rb` in this PR (so that they won't cause troubles for other arrow developers), and we could add them back once the related failures are resolved